lametro/management/commands/generate_tag_analytics.py

```diff
@@ -1,34 +1,114 @@
 import csv
+from io import StringIO
+from datetime import datetime
 from tqdm import tqdm
+
+from django.conf import settings
 from django.core.management.base import BaseCommand
 
+from googleapiclient.http import MediaIoBaseUpload
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from google.oauth2 import service_account
+
 from lametro.models import LAMetroBill
+from lametro.exceptions import DriveBuildError, UploadError
 
 
 class Command(BaseCommand):
 
-    help = "This command produces a CSV file that lists each Board Report's tags."
+    help = "This command produces a CSV file that lists each Board Report's tags and \
+        uploads it to the 'LA Metro Reports' folder in Google Drive."
 
     def handle(self, *args, **options):
 
-        with open('tag_analytics.csv', 'w', newline='') as f:
-            writer = csv.writer(f, delimiter=',')
-            writer.writerow(
-                ('File ID',
-                 'Identifier'
-                 'Title',
-                 'Last Action Date',
-                 'Tag Classification',
-                 'Tag')
+        csv_string = self.generate_tag_analytics()
+        date = datetime.today().strftime('%m_%d_%y')
+        output_file_name = f'{date}_tag_analytics.csv'
+
+        file_metadata = {
+            'name': output_file_name,
+            'parents': [settings.REMOTE_ANALYTICS_FOLDER]
+        }
+
+        try:
+            drive_service = self.get_google_drive()
+            file = self.upload_file_bytes(drive_service, csv_string, file_metadata)
+            self.stdout.write(
+                self.style.SUCCESS(f'Successfully uploaded {output_file_name}!')
+            )
+        except DriveBuildError as e:
+            self.stdout.write(
+                self.style.ERROR(f'Unable to connect to Google Drive!')
             )
+            raise
+        except HttpError as e:
+            self.stdout.write(
+                self.style.ERROR(f'HTTP Error!')
+            )
+            raise
```

Comment:
This note seems redundant, since you go on to raise the HTTP error.

Reply:
Good point.